icechunk/src/strategies.rs

```diff
@@ -180,7 +180,7 @@ prop_compose! {
             S3Compatible(_) => VirtualChunkContainer::new("s3://somebucket/".to_string(),store).unwrap(),
             S3(_) => VirtualChunkContainer::new("s3://somebucket/".to_string(),store).unwrap(),
             Gcs(_) => VirtualChunkContainer::new("gcs://somebucket/".to_string(),store).unwrap(),
-            Azure(_) => VirtualChunkContainer::new("az://somebucket/".to_string(),store).unwrap(),
+            Azure{..} => VirtualChunkContainer::new("az://somebucket/".to_string(),store).unwrap(),
```

Comment:
confused about why this works since we have `Azure(HashMap<String, String>)` above.

EDIT: There are a bunch of similar changes below too